src/doc/trpl/strings.md

```diff
@@ -49,6 +49,18 @@ fn main() {
 }
 ```
 
+This coercion does not happen for functions that accept one of `&str`’s traits
+instead of `&str`. For example, [`TcpStream::connect`][connect] has a parameter
+of type `ToSocketAddrs`. A `&str` is okay but a `String` must be explicitly
+converted using `&*`.
+
+```rust
```

Comment:
this should probably be tagged `no_run` as well, because it won't work, since there's no server, yeah?